tsdb/chunkenc/chunk_test.go

```diff
@@ -217,3 +217,40 @@ func benchmarkAppender(b *testing.B, newChunk func() Chunk) {
 
 	fmt.Println("num", b.N, "created chunks", len(chunks))
 }
+
+func TestXor(t *testing.T) {
+	timestamp := int64(1234123324)
+	v := 83665193758603.123
+	num := 10
+	var samples []pair
+	for i := 0; i < num; i++ {
+		timestamp += int64(1000)
+		v += float64(100)
+		samples = append(samples, pair{t: timestamp, v: v})
+	}
+
+	chunk := NewXORChunk()
+	a, err := chunk.Appender()
+	testutil.Ok(t, err)
+	for _, p := range samples {
+		a.Append(p.t, p.v)
+	}
+
+	res := make([]pair, num)
+	var it Iterator
+	it = chunk.Iterator(it)
+	for it.Next() {
+		t, v := it.At()
+		res = append(res, pair{t: t, v: v})
+	}
+	if it.Err() != io.EOF {
+		testutil.Ok(t, it.Err())
+	}
+
+	for i := 0; i < len(samples); i++ {
+		s := samples[i]
+		r := res[i]
+		testutil.Equals(t, s.t, r.t, "test compact timestamp fail ,delta of delta", s, r)
+		testutil.Equals(t, s.v, r.v, "test compact value fail ,xor", s, r)
```

Comment:
```suggestion
		testutil.Equals(t, s.t, r.t, "test compact timestamp fail (delta of delta)")
		testutil.Equals(t, s.v, r.v, "test compact value fail (xor)")
```

testutil.Equal will show the right values,

Would testutil.Equals(t, s, r) work, too?

Reply:
done,thanks for your review.